src/main/java/seedu/duke/DukeSession.java

```diff
@@ -47,7 +53,7 @@ public DukeControlFlow getControlFlow() {
     /**
      * Runs the read, evaluate, print loop for Duke.
      */
-    public void runDuke() {
+    public void runDuke() throws DukeException {
```

Comment:
Maybe it would be a good idea to handle the exception instead of just throwing it up into main (which also results in the exiting of the program)